tests/test_downloadermiddleware_retry.py

```diff
@@ -103,6 +103,41 @@ def _test_retry_exception(self, req, exception):
         req = self.mw.process_exception(req, exception, self.spider)
         self.assertEqual(req, None)
 
+    def test_different_retry(self):
+        
+        req = Request('http://www.scrapytest.org/invalid_url', meta={'max_retry_times': 3})
+
+        # SETINGS: meta(max_retry_times) = 3, RETRY_TIMES = 2
+        self._test_retry(req, DNSLookupError('foo'), 3)
+
+        req2 = Request('http://www.scrapytest.org/invalid_url')
+        
```

Comment:
Great! Creating a new testcase class seems like a very good option. Working on it!